tests/experimental/test_lightning_qubit_2.py

```diff
@@ -0,0 +1,330 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""General tests for Lightning Qubit 2."""
+import pytest
+
+from pennylane import numpy as np
+
+import pennylane as qml
+from pennylane.devices.experimental import ExecutionConfig
+from pennylane_lightning.experimental import LightningQubit2
+
+from pennylane_lightning.experimental.lightning_qubit_2 import CPP_BINARY_AVAILABLE
+
+
+@pytest.mark.skipif(not CPP_BINARY_AVAILABLE, reason="Lightning binary required")
+def test_name():
+    """Tests the name of LightningQubit2."""
+    assert LightningQubit2().name == "lightning.qubit.2"
+
+
+@pytest.mark.skipif(CPP_BINARY_AVAILABLE, reason="Only when there is no binary")
+def test_name_no_binary():
+    """Tests that we are offloading to the default qubit."""
+    assert LightningQubit2().name == "default.qubit.2"
+
+
+@pytest.mark.skipif(
+    not hasattr(np, "complex256"), reason="Numpy only defines complex256 in Linux-like system"
+)
+def test_create_device_with_unsupported_dtype():
+    with pytest.raises(TypeError, match="Unsupported complex Type:"):
+        LightningQubit2(c_dtype=np.complex256)
+
+
+@pytest.mark.skipif(not CPP_BINARY_AVAILABLE, reason="Lightning binary required")
+def test_no_jvp_functionality():
+    """Test that jvp is not supported on LightningQubit2."""
+    dev = LightningQubit2()
+
+    assert not dev.supports_jvp(ExecutionConfig())
+
+    with pytest.raises(NotImplementedError):
+        dev.compute_jvp(qml.tape.QuantumScript(), (10, 10))
+
+    with pytest.raises(NotImplementedError):
+        dev.execute_and_compute_jvp(qml.tape.QuantumScript(), (10, 10))
+
+
+@pytest.mark.skipif(not CPP_BINARY_AVAILABLE, reason="Lightning binary required")
+def test_no_vjp_functionality():
+    """Test that vjp is not supported on LightningQubit2."""
```

Comment:
I assume this comes in one of the follow-up PRs?

Reply:
This was on the nice-to-have list. 
I didn't have time to do that last quarter.
Let me know, if you want me to implement it.